src/System.Linq.Expressions/tests/Constant/ConstantTests.cs

```diff
@@ -276,7 +297,17 @@ public static void CheckUShortConstantTest(bool useInterpreter)
         [Theory, ClassData(typeof(CompilationTypes))]
         public static void CheckTypeConstantTest(bool useInterpreter)
         {
-            foreach (Type value in new Type[] { null, typeof(int), typeof(Func<string>) })
+            foreach (Type value in new[]
+            {
+                null,
+                typeof(int),
+                typeof(Func<string>),
+                typeof(List<>).GetGenericArguments()[0],
+                GetTypeBuilder(),
```

Comment:
Yes. `ArgumentException` with a message of "Type must be a runtime Type object."